includes/class-sensei.php

```diff
@@ -580,27 +575,67 @@ public function install() {
 	} // End install()
 
 	/**
-	 * Check for plugin updates.
+	 * Checks for plugin update tasks and ensures the current version is set.
 	 *
 	 * @since 2.0.0
 	 */
 	public function update() {
-		if ( ! version_compare( $this->version, get_option( 'sensei-version' ), '>' ) ) {
+		$current_version = get_option( 'sensei-version' );
+		$is_new_install  = ! $current_version && ! get_option( 'sensei-settings', get_option( 'woothemes-sensei-settings' ) );
```

Comment:
Is the `get_option( 'sensei-settings' )` safe? I tested with a fresh install and the `sensei-settings` was created only after go to settings and click in "Save Changes".

Reply:
You're right--it isn't 100% safe, but it is just an additional safeguard in addition to `sensei-version`. Actually, when I checked it out, I remembered I changed this quite a bit in v3 so I'm not sure how reliable it is.

I was reluctant to add a query here because this is ran on every page load forever, however, I've switched it to a course check and put it after the `sensei-version` check. With that, it should only have to ping the database once and then it'll set `sensei-version` and never do it again.

efcba53a3f99ed956d5a5350d88e129443b206f3